engine/api/src/main/java/io/deephaven/engine/table/updateBySpec/EmaSpec.java

```diff
@@ -1,61 +1,46 @@
 package io.deephaven.engine.table.updateBySpec;
 
+import io.deephaven.annotations.SimpleStyle;
 import io.deephaven.engine.table.EmaControl;
 import io.deephaven.util.type.TypeUtils;
+import org.immutables.value.Value.Parameter;
+import org.immutables.value.Value.Immutable;
 import org.jetbrains.annotations.NotNull;
 import org.jetbrains.annotations.Nullable;
 
 /**
  * A {@link UpdateBySpec} for performing an Exponential Moving Average across the specified columns
  */
-public class EmaSpec implements UpdateBySpec {
-    private final String timestampCol;
-    private final long timeScaleUnits;
-    private final EmaControl control;
-
+@Immutable
+@SimpleStyle
+public abstract class EmaSpec implements UpdateBySpec {
     public static EmaSpec ofTime(@NotNull final EmaControl control,
             @NotNull final String timestampCol,
             long timeScaleNanos) {
-        return new EmaSpec(control, timestampCol, timeScaleNanos);
+        return ImmutableEmaSpec.of(control, timestampCol, timeScaleNanos);
     }
 
     public static EmaSpec ofTicks(@NotNull EmaControl control, long tickWindow) {
-        return new EmaSpec(control, null, tickWindow);
+        return ImmutableEmaSpec.of(control, null, tickWindow);
     }
```

Comment:
I see 3 different variable names: `timeScaleNanos`, `tickWindow` and `timeScaleUnits`. Are these all the same thing?